File: bookStoreApp/src/app/books/models/book.model.ts
import { AuthorModel } from "./author.model";
import { PriceModel } from "./price.model";

/* export interface BookModel
{
    id:number,
    title:string,
    totalPages:number,
    author:string,
    price:PriceModel
} */

export class BookModel
{
    public id:number;
    public title:string;
    public totalPages:number;
    public author:string;
    public price:PriceModel;
    public isPublished:boolean;
    public datePublished:Date;
    public authors:AuthorModel[];
}

File: BookStoreApi/BookStoreApi/Models/BookModel.cs
namespace BookStoreApi.Models
{
    public class BookModel
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public int TotalPages { get; set; }
        public PriceModel Price { get; set; }
        public bool IsPublished { get; set; }
        public DateTime DatePublished { get; set; }
        public List<AuthorModel> Authors { get; set; }
    }
}
